packages/airnode-adapter/src/response-processing/array-type.ts

```diff
@@ -0,0 +1,54 @@
+import { ResponseType, BaseResponseType, baseResponseTypes } from '../types';
+
+// Numeric types could be multiplied by the "_times" reserved parameter
+export function isNumericType(type: ResponseType): type is 'uint256' | 'int256' {
+  return type === 'int256' || type === 'uint256';
+}
+
+export interface ParsedArrayType {
+  readonly baseType: BaseResponseType;
+  readonly dimensions: number;
+}
+
+export function parseArrayType(type: ResponseType): ParsedArrayType | null {
+  if (!type.includes('[')) return null;
+
+  const typeMatch = type.match(/^([^\[]+)(.*)$/);
+  if (!typeMatch || !baseResponseTypes.includes(typeMatch[1] as any)) return null;
+  const baseType = typeMatch[1] as BaseResponseType;
+
+  // eslint-disable-next-line functional/no-let
+  let dimensionsString = typeMatch[2];
+  // eslint-disable-next-line functional/no-let
+  let dimensions = 0;
+  // eslint-disable-next-line functional/no-loop-statement
```

Comment:
The ESLint functional plugin is bothering me more and more. The only part of it that I *really* like is the enforced immutability (with opt-out when needed)

Reply:
Tell me something about it... I think in this function I wrote more eslint disable lines than logic. I've bumped the priority of https://api3dao.atlassian.net/browse/AN-190